src/Utils/Hooks/useCountdownTimer.ts

```diff
@@ -1,26 +1,50 @@
 import { useState, useEffect } from "react"
-import { differenceInHours, differenceInMinutes } from "date-fns"
+import {
+  differenceInHours,
+  differenceInMinutes,
+  differenceInSeconds,
+} from "date-fns"
 
 const IMMINENT_TIME = 5 // hours
 
-const calculateTime = (endTime: string) => {
+const calculateTime = (endTime: string, includeSeconds: boolean) => {
   const now = new Date()
   const expiration = new Date(endTime)
-  const timeDiff = differenceInMinutes(expiration, now)
 
-  const ONE_DAY = 1440 // 24 hours * 60 minutes
-
-  const days = Math.floor(timeDiff / ONE_DAY)
-  const hours = Math.floor((timeDiff % ONE_DAY) / 60)
-  const minutes = Math.floor(timeDiff % 60)
-
-  if (timeDiff <= 0) {
-    return "Expired"
-  }
-  if (timeDiff >= ONE_DAY) {
-    return `${days}d ${hours}h`
+  if (!includeSeconds) {
+    const timeDiff = differenceInMinutes(expiration, now)
+
+    const ONE_DAY = 1440 // 24 hours * 60 minutes
+
+    const days = Math.floor(timeDiff / ONE_DAY)
+    const hours = Math.floor((timeDiff % ONE_DAY) / 60)
+    const minutes = Math.floor(timeDiff % 60)
+
+    if (timeDiff <= 0) {
+      return "Expired"
+    }
+    if (timeDiff >= ONE_DAY) {
+      return `${days}d ${hours}h`
+    }
+    return `${hours}h ${minutes}m`
+  } else {
+    const timeDiff = differenceInSeconds(expiration, now)
+
+    const ONE_DAY = 86400 // 24 hours * 60 minutes * 60 seconds
+
+    const days = Math.floor(timeDiff / ONE_DAY)
+    const hours = Math.floor((timeDiff % ONE_DAY) / (60 * 60))
+    const minutes = Math.floor((timeDiff % (60 * 60)) / 60)
+    const seconds = Math.floor(timeDiff % 60)
```

Comment:
Instead of duplicating this logic with minutes vs seconds, why not always calculate it with seconds (and don't output the seconds if `includeSeconds` is false)?

Reply:
Thanks, @iskounen . That's a great idea. Updated the logic.